frontend/src/components/Common/StudentDetailsDialog.vue

```diff
@@ -4,7 +4,7 @@
       variant="plain"
       color="error"
       v-ripple
-      @click="studentDialog = !studentDialog"
+      @click="studentDialog = !studentDialog && (showHideAdopt = true)"
```

Comment:
Rename showHideAdopt so it's more clear since show and hide are contradictory verbs.